components/ws-proxy/pkg/proxy/pass.go

```diff
@@ -81,10 +86,16 @@ func proxyPass(config *RouteHandlerConfig, resolver targetResolver, opts ...prox
 	return func(w http.ResponseWriter, req *http.Request) {
 		targetURL, err := h.TargetResolver(config.Config, req)
 		if err != nil {
-			log.Errorf("Unable to resolve targetURL: %s", req.URL.String())
+			log.WithError(err).Errorf("Unable to resolve targetURL: %s", req.URL.String())
 			return
 		}
 
+		// If the targetURL needs a path prefix, it will have set that as path.
```

Comment:
I'm not sure I understand the comment: Is this necessary for cases where we want to drop a path prefix (the `/ide` case), because `NewSingleHostReverseProxy` only supports appending?

Reply:
That's a good point. Originally I assumed that the `SingleHostReverseProxy` would not support paths in the target URL; an assumption corroborated by this change which made things work.

Turns out the proxy does support paths, so the issue must be elsewhere and this change just makes it. I'll look into it once the dev-staging DNS issue is resolved.

Reply:
Turns out there was a slash (`/`) missing at the beginning of the blobserve path. Now it works without this hack, using the proxy alone.